### DISCUSSION TITLE
Need help to figure out how to create new automation using either HomeAssistantAPI or REST API

### DISCUSSION THREAD

**Discussion by alan-eero** [GrandMoff100/HomeAssistantAPI#190] [Q&A]
My goal is to be able to create/delete automations using python scripts. 
An automation such that trigger + condition = action. 

using the front-end, this can be done with: settings --> automation & scenes --> create automation. 

I tried POST a json to url .../config/automation/config/<random-id>, 
but does not work. 

```
automation_json = {
        "id": "1234567890",
        "alias": "dummy_automation",
        "description": "dummy dummy",
        "trigger": [
            {
                "service": "light.toggle",
                "target": {
                    "entity_id": "switch.mini_smart_wi_fi_plug_switch_3"
                },
                "metadata": {}
            }
        ],
        "condition": [],
        "action": [
            {
                "service": "switch.toggle",
                "target": {
                    "entity_id": "switch.mini_smart_wi_fi_plug_switch_3"
                },
                "metadata": {}
            }
        ],
        "mode": "single"
    }
```

Thank you very much in advance for your valuable help.

**Comment by GrandMoff100** [comment]
Do you need to create them with HTTP requests specifically? Or is SSH an option (like a Python ssh-wrapper library or bash script)?

**Reply by alan-eero** [reply]
thanks for your response.
right now, any option will be good. Python ssh-wrapper sounds like a good idea. 
I just installed the advanced SSH & terminal add-on. I can ssh into the HomeAssistant host. I see there is a "ha" command line interface available. Please advice if you already have identified which commands can help me to accomplish the goal.

**Comment by sanzoghenzo** [comment]
Necroposting this since I'm researching how to do this right now and I just discovered this wonderful package...

I found this discussion on the HA community:

https://community.home-assistant.io/t/rest-api-docs-for-automations/119997/10

Basically add a random id to the url you were trying: `/config/automation/config/<random-id>` 

Although undocumented, it would be nice to have this added to the library.

- The id of an existing automation can be found in the `attributes["id"]` its state (`entity_id`s of automations starts with "automation.")
- GET /config/automation/config/<id> to retrieve the existing automation, it is basically the json format of what you see in the yaml editor
- POST /config/automation/config/<id> will create (new id) or update (existing id) an automation
